source/h2_connection.c

```diff
@@ -1247,16 +1338,8 @@ struct aws_h2err s_decoder_on_data_begin(
         auto_window_update = payload_len;
     }
 
-    if (auto_window_update != 0) {
-        if (s_connection_send_update_window(connection, auto_window_update)) {
-            return aws_h2err_from_last_error();
-        }
-        CONNECTION_LOGF(
-            TRACE,
-            connection,
-            "Automatically updating connection window by %" PRIu32 "(%" PRIu32 " due to padding).",
-            auto_window_update,
-            total_padding_bytes);
+    if (s_connection_send_update_window_if_needed(connection, auto_window_update)) {
+        return aws_h2err_from_last_error();
```

Comment:
trivial: as I mentioned elsewhere, either don't call this function when auto_window_update is 0, or have the function early-out when it's 0